File: Basic/cordapp-example/contracts/src/test/java/net/corda/samples/example/contracts/StateTests.java
package net.corda.samples.example.contracts;

import net.corda.samples.example.states.IOUState;
import org.junit.Test;

import static org.junit.Assert.assertSame;

public class StateTests {
    @Test
    public void hasAmountFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        IOUState.class.getDeclaredField("value");
        // Is the message field of the correct type?
        assertSame(IOUState.class.getDeclaredField("value").getType(), Integer.class);
    }
}

File: kotlin-samples/Features/notarychange-iou/contracts/src/test/kotlin/net/corda/samples/notarychange/contracts/StateTests.kt
package net.corda.samples.notarychange.contracts

import net.corda.samples.notarychange.states.IOUState
import net.corda.testing.node.MockServices
import org.junit.Test

class StateTests {
    private val ledgerServices = MockServices()
    @Test
    @Throws(NoSuchFieldException::class)
    fun hasAmountFieldOfCorrectType() {
        // Does the message field exist?
        IOUState::class.java.getDeclaredField("value")
        assert(IOUState::class.java.getDeclaredField("value").getType() == Int::class.javaPrimitiveType)
    }
}